WordPress/openverse
# Re-add Storybook

## Fixes



Fixes #4728 by @obulat

## Description


This PR adds the nightly version of Storybook to the app.

While it is not ideal to use a nightly version, especially with still existing issues, having Storybook is very important for the development of the interface features such as the Dark mode.

### Existing issues in Nuxt storybook module:
- [Cannot read properties of undefined (reading 'cdnURL')](https://github.com/nuxt-modules/storybook/issues/753#top) is still a problem that causes the images (home page gallery, error page images, local images for the image cell stories) to not render.

### Other notable changes
1. Most stories were converted to TS. When the component rendered in the Story has extra props that do not exist on the component (eg., in `VButton` story, we also have a `variants` prop for the Variants stories that does not exist in the `VButton` component), I left the component in JS because of difficulties with typing
2. All stories were converted to use the `h => ()` render functions instead of the `template` string because of the Storybook module error (no vue compiler available, TODO: Add the error message)
3. Some props were simplified.
4. Added emits to `VButton`

## Testing Instructions

0. Install dependencies `ov just install` and run storybook `ov just frontend/run storybook`.
1. View Storybook. While there are some issues logged in the console, all stories should display correctly.
6. View the CI 

## Checklist


      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
